vm-virtio/src/iommu.rs

```diff
@@ -759,16 +759,16 @@ struct Mapping {
 
 pub struct IommuMapping {
     // Domain related to an endpoint.
-    endpoints: Arc<RwLock<BTreeMap<u32, u32>>>,
+    endpoints: Arc<Mutex<BTreeMap<u32, u32>>>,
     // List of mappings per domain.
-    mappings: Arc<RwLock<BTreeMap<u32, BTreeMap<u64, Mapping>>>>,
+    mappings: Arc<Mutex<BTreeMap<u32, BTreeMap<u64, Mapping>>>>,
 }
 
 impl DmaRemapping for IommuMapping {
     fn translate(&self, id: u32, addr: u64) -> std::result::Result<u64, std::io::Error> {
         debug!("Translate addr 0x{:x}", addr);
-        if let Some(domain) = self.endpoints.read().unwrap().get(&id) {
-            if let Some(mapping) = self.mappings.read().unwrap().get(domain) {
+        if let Some(domain) = self.endpoints.lock().unwrap().get(&id) {
+            if let Some(mapping) = self.mappings.lock().unwrap().get(domain) {
```

Comment:
I *think* the deadlock is here (just from inspection.)

Reply:
Thank you for the hint.